openff/toolkit/topology/molecule.py

```diff
@@ -695,7 +696,13 @@ def bond_order(self):
 
     @bond_order.setter
     def bond_order(self, value):
-        self._bond_order = value
+        if isinstance(value, int):
+            self._bond_order = value
+        else:
+            raise InvalidBondOrderError(
+                "Non-integer bond order passed to `Bond.bond_order` setter.  Consider using "
+                "`Bond.fractional_bond_order` for fractional/partial bond orders."
```

Comment:
I think this may mislead people who intend to set an "aromatic" bond order of 1.5 to set the partial bond order to 1.5 instead of passing in a kekulization of their intended structure. 

```suggestion
                "Only integer bond orders may be passed to `Bond.bond_order` setter. "
                "For aromatic bonds, instead kekulize the input structure and use "
                "the resulting integer bond orders. If performing partial bond "
                "order-based parameter interpolation, consider using "
                "`Bond.fractional_bond_order`."
```